### PR TITLE
mv contrib/cmd tests/cmd (except memfd-bind)

### PR DESCRIPTION
**Pull request by AkihiroSuda**
The following commands are moved from `contrib/cmd` to `tests/cmd`:
- fs-idmap
- pidfd-kill
- recvtty
- remap-rootfs
- sd-helper
- seccompagent

So as to clarify that distro maintainers should not include these binaries in their package.

ref: https://bugs.launchpad.net/ubuntu/+source/runc/+bug/2076981

### CODE REVIEW

**Review by rata — DISMISSED** [review]
This LGTM, but do you think this will be enough?

Maybe we can change the Makefile in some way, so those don't get build except for tests? Or print in "make all" some warning that test binaries are created, for a release you should use "make XXX"?

**Comment by AkihiroSuda** [comment]
Updated `make all` to print a note.

File: Makefile [hunk 1]
```diff
@@ -79,6 +79,8 @@ runc-bin: runc-dmz
 
 .PHONY: all
 all: runc recvtty sd-helper seccompagent fs-idmap memfd-bind pidfd-kill remap-rootfs
+	@echo "*Note*: Only the runc binary should be installed to user environments."
```
Comment (lifubang) [inline Makefile:82]: Please add a new line after `*Note*: `, or add 8 spaces before `Other binaries...`?
I think it will looks more beautiful.

**Comment by kolyshkin** [comment]
In fact, we have 8 entries in there.
* 7 of them are used by tests/integration, and should probably be moved there (say to `tests/cmd/`). The only other thing to do is to fix the link to recvtty from docs/terminal.md.
* 1 (`memfd-bind`) can be kept as-is, its documentation is adequately telling distro maintainers what to do with it.

File: Makefile (hunk 1 above)
Comment (kolyshkin) [reply Makefile:82]: Or maybe don't make make more talkative at all.

**Comment by thaJeztah** [comment]
Would moving things to `internal/` be something we should consider? It makes it crystal clear "for own use"?

**Comment by kolyshkin** [comment]
> Would moving things to `internal/` be something we should consider? It makes it crystal clear "for own use"?

When writing [the above comment](https://github.com/opencontainers/runc/pull/4377#issuecomment-2290936966) yesterday, I was going to suggest `internal/cmd` but then realized these are mostly used for tests, and we have a directory for that already.

I think `tests/cmd` is a good enough path to figure out these binaries are for tests and thus should not be included into runc package.

Also, two of the programs (recvtty and seccomagent) are also used as code examples, so they are not entirely "internal".

File: tests/integration/helpers.bash [hunk 2]
```diff
@@ -13,12 +13,12 @@ eval "$IMAGES"
...
-SECCOMP_AGENT="${INTEGRATION_ROOT}/../../contrib/cmd/seccompagent/seccompagent"
-FS_IDMAP="${INTEGRATION_ROOT}/../../contrib/cmd/fs-idmap/fs-idmap"
-PIDFD_KILL="${INTEGRATION_ROOT}/../../contrib/cmd/pidfd-kill/pidfd-kill"
-REMAP_ROOTFS="${INTEGRATION_ROOT}/../../contrib/cmd/remap-rootfs/remap-rootfs"
+RECVTTY="${INTEGRATION_ROOT}/../../tests/cmd/recvtty/recvtty"
+SD_HELPER="${INTEGRATION_ROOT}/../../tests/cmd/sd-helper/sd-helper"
+SECCOMP_AGENT="${INTEGRATION_ROOT}/../../tests/cmd/seccompagent/seccompagent"
+FS_IDMAP="${INTEGRATION_ROOT}/../../tests/cmd/fs-idmap/fs-idmap"
+PIDFD_KILL="${INTEGRATION_ROOT}/../../tests/cmd/pidfd-kill/pidfd-kill"
+REMAP_ROOTFS="${INTEGRATION_ROOT}/../../tests/cmd/remap-rootfs/remap-rootfs"
```
Comment (kolyshkin) [inline tests/integration/helpers.bash:21]: nit: all these should now be like `"${INTEGRATION_ROOT}/../cmd/remap-rootfs/remap-rootfs"`

File: tests/integration/seccomp-notify.bats [hunk 3]
```diff
@@ -214,7 +214,7 @@ function scmp_act_notify_template() {
 @test "runc run [seccomp] (SCMP_ACT_NOTIFY example config)" {
 	# Run the script used in the seccomp agent example.
 	# This takes a bare config.json and modifies it to run an example.
-	"${INTEGRATION_ROOT}/../../contrib/cmd/seccompagent/gen-seccomp-example-cfg.sh"
+	"${INTEGRATION_ROOT}/../../tests/cmd/seccompagent/gen-seccomp-example-cfg.sh"
```
Comment (kolyshkin) [inline tests/integration/seccomp-notify.bats:217]:
```suggestion
	"${INTEGRATION_ROOT}/../cmd/seccompagent/gen-seccomp-example-cfg.sh"
```

File: tests/integration/helpers.bash (hunk 2 above)
Comment (kolyshkin) [inline tests/integration/helpers.bash:21]: IOW
```suggestion
RECVTTY="${INTEGRATION_ROOT}/../cmd/recvtty/recvtty"
SD_HELPER="${INTEGRATION_ROOT}/../cmd/sd-helper/sd-helper"
SECCOMP_AGENT="${INTEGRATION_ROOT}/../cmd/seccompagent/seccompagent"
FS_IDMAP="${INTEGRATION_ROOT}/../cmd/fs-idmap/fs-idmap"
PIDFD_KILL="${INTEGRATION_ROOT}/../cmd/pidfd-kill/pidfd-kill"
REMAP_ROOTFS="${INTEGRATION_ROOT}/../cmd/remap-rootfs/remap-rootfs"
```
(not tested)

Comment (AkihiroSuda) [reply tests/integration/helpers.bash:21]: Can be another PR?

**Review by kolyshkin — APPROVED** [review]
lgtm

**Review by rata — APPROVED** [review]
LGTM, thanks!

**Pull request merged**